cvat-ui/src/components/analytics-page/consensus-analytics-page.tsx

```diff
@@ -0,0 +1,220 @@
+    const tab = window.location.hash.slice(1) as ConsensusAnalyticsTabs;
+    return Object.values(ConsensusAnalyticsTabs).includes(tab) ? tab : ConsensusAnalyticsTabs.OVERVIEW;
+}
+
+type InstanceType = 'task';
+
+function TaskConsensusAnalyticsPage(): JSX.Element {
+    const dispatch = useDispatch();
+
+    const requestedInstanceType: InstanceType = 'task';
+    const requestedInstanceID = +useParams<{ tid: string }>().tid;
+
+    const [activeTab, setTab] = useState(getTabFromHash());
+    const [instanceType, setInstanceType] = useState<InstanceType | null>(null);
+    const [instance, setInstance] = useState<Task | null>(null);
+    const [fetching, setFetching] = useState(true);
+    const isMounted = useIsMounted();
+    const consensusSettings = useSelector((state: CombinedState) => state.consensus?.consensusSettings);
+
+    const onTabKeyChange = useCallback((key: string): void => {
+        setTab(key as ConsensusAnalyticsTabs);
+    }, []);
+
+    const receiveInstance = async (type: InstanceType, id: number): Promise<void> => {
+        let receivedInstance: Task | null = null;
+
+        try {
+            switch (type) {
+                case 'task': {
+                    [receivedInstance] = await core.tasks.get({ id });
+                    break;
+                }
+                default:
+                    return;
+            }
+
+            if (isMounted()) {
+                setInstance(receivedInstance);
+                setInstanceType(type);
+            }
+        } catch (error: unknown) {
+            notification.error({
+                message: `Could not receive requested ${type}`,
+                description: `${error instanceof Error ? error.message : ''}`,
+            });
+        }
+    };
+
+    useEffect(() => {
+        if (Number.isInteger(requestedInstanceID) && ['project', 'task', 'job'].includes(requestedInstanceType)) {
+            setFetching(true);
+            Promise.all([
+                receiveInstance(requestedInstanceType, requestedInstanceID),
+            ]).finally(() => {
+                if (isMounted()) {
+                    setFetching(false);
+                }
+            });
+        } else {
+            notification.error({
+                message: 'Could not load this page',
+                description: `Not valid resource ${requestedInstanceType} #${requestedInstanceID}`,
+            });
+        }
+
+        return () => {
+            if (isMounted()) {
+                setInstance(null);
+            }
+        };
+    }, [requestedInstanceType, requestedInstanceID]);
+
+    function handleError(error: Error): void {
+        notification.error({
+            description: error.toString(),
+            message: 'Could not fetch consensus settings.',
+        });
+    }
+
+    useEffect(() => {
+        window.addEventListener('hashchange', () => {
+            const hash = getTabFromHash();
+            setTab(hash);
+        });
```

Comment:
Any event listener should be destroyed in useEffect's return. Otherwise this is memory leak

Reply:
Like this. Correct?

```js
return () => {
    window.removeEventListener('hashchange', () => {
        setTab(getTabFromHash());
    });
}
```

I guess I also make this change in `cvat-ui/src/components/analytics-page/analytics-page.tsx`